nxdl/nxdl_to_hdf5.py

```diff
@@ -1536,6 +1559,20 @@ def symbol_args_to_dict(arg_lst):
 
     return(dct)
 
+def process_nxdl(class_nm, def_subdir):
+    '''
+    given the class name and the destination directory, parse the nxdl file and produce an hdf5 file
+    as well as example scripts using h5py and nexusformat
+    '''
+    if class_nm.find('.nxdl.xml') > -1:
+        class_nm = class_nm.replace('.nxdl.xml', '')
+        build_class_db(def_subdir, desired_class=class_nm,
+                       defdir=def_dir, sym_args_dct=sym_args_dct,
+                       report_symbols_only=report_symbols_only)
+        dest_dir = pathlib.PurePath(pathlib.os.getcwd(), '..', 'autogenerated_examples', 'nxdl', def_subdir)
+        make_class_as_nf_file(class_nm, dest_dir, symbol_dct=sym_args_dct)
+
+
 if __name__ == '__main__':

```

Comment:
This block of code is sufficiently complicated that it can be its own function (such as `main()`).  By moving into a separate function, the code can be made into an entry point that can be called from other code or cited in a `setup.py` file.

Reply:
sorry Pete but I don't think that I am 100% understanding you.  I can see down the road that this code could be used to return a dict version of an nxdl file to be used as a template for a data collection app to use for the data but I think I would need to think about it more (like a separate branch) as the code at the moment is pretty specific to cranking out h5 files and python scripts, and the time I would need (to implement what I think you are suggesting) exceeds the current amount I have available to put towards this.

Reply:
see my new branch [29-prj2](https://github.com/nexusformat/exampledata/tree/29-prj2) and PR #32

Reply:
The entry point aspect would be very similar to a [change for the NeXusOntology described](https://github.com/nexusformat/NeXusOntology/pull/3#issuecomment-994952715) earlier today.

Reply:
It's used first when you want to define an application from the `setup.py` file.  (You don't have one of those but it _would_ make anyone's use of this code much easier).

Reply:
I will defer to your expertise on this regarding setup.py etc, its not something I have a lot of experience with so if you think its a good idea that's good enough for me.